src/oceans/components/common/Guide.jsx

```diff
@@ -1,29 +1,82 @@
-import React from 'react'
-import Radium from "radium";
-import Typist from "react-typist";
-
-import {getState, setState} from "@ml/oceans/state";
-import guide from "@ml/oceans/models/guide";
-import soundLibrary from "@ml/oceans/models/soundLibrary";
-import styles from "@ml/oceans/styles";
-import colors from "@ml/oceans/styles/colors";
-import I18n from "@ml/oceans/i18n";
-import {Button} from "@ml/oceans/components/common";
-import arrowDownImage from "@public/images/arrow-down.png";
+import React from 'react';
+import Radium from 'radium';
+import Typist from 'react-typist';
+
+import {getState, setState} from '@ml/oceans/state';
+import guide from '@ml/oceans/models/guide';
+import soundLibrary from '@ml/oceans/models/soundLibrary';
+import styles from '@ml/oceans/styles';
+import colors from '@ml/oceans/styles/colors';
+import I18n from '@ml/oceans/i18n';
+import {Button} from '@ml/oceans/components/common';
+import arrowDownImage from '@public/images/arrow-down.png';
+import {
+  startTextToSpeech,
+  stopTextToSpeech,
+  hasTextToSpeechVoices
+} from '@ml/utils/TextToSpeech';
+
+// A timer used for playing typing sounds.
+let guideTypingTimer = undefined;
+
+// Whether text to speech has ever been successfully
+// started via a user click.
+let textToSpeechStartedViaClick = false;
```

Comment:
Would `hasTextToSpeechStarted` or `hasTextToSpeechStartedByClick` be an easier-to-interpret variable name here?

Reply:
Updated to `hasTextToSpeechStartedByClick`.